pkgs/os-specific/linux/zmalllinugz/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, fetchurl, coreutils, flex, bison, libelf, bc, hostname, perl }:
+
+
+let  
+  version = "5.3.8";
+  src = fetchurl {
+    url = "mirror://kernel/linux/kernel/v5.x/linux-${version}.tar.xz";
+    sha256 = "0jb6yya9yx4z52p5m32dqj0kgc6aaz9df8mvq0hzy40bqb3czwvq";
+  };
+  config = ./zmalllinugz.config;
+
+in
+stdenv.mkDerivation rec {
+  name = "zmalllinugz";
+
+  srcs = [ src config ];
+
+  dontConfigure=true;
+  buildInputs = [ coreutils flex bison libelf bc hostname perl];
+  
+  unpackPhase = ''
+    array=($srcs)
+    kernel_tar_gz=''${array[0]}
+    zmalllinugz_config=''${array[1]}
+    tar -xf $kernel_tar_gz
+    sourceRoot=linux-${version}
+    cp $zmalllinugz_config $sourceRoot/.config
+
+  '';
+  buildPhase = ''
+    make -j $NIX_BUILD_CORES all
+  '';
+  installPhase = ''
+    mkdir -p $out/kernel 
+    cp arch/x86/boot/bzImage $out/kernel
+    cp .config $out/kernel
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Smallest possible linux kernel for KVM";
+    longDescription = ''
```

Comment:
Seems like this is a package just for yourself? In this case I don't think it belongs in Nixpkgs.

Reply:
Well any package at the beginning is for "oneself". Hopefully it will be useful for others for any project that needs a simple kernel for VM's. As mentioned in the description this is an enabler for another project I am working on, but atm did not want to disclose to much. The build will be important to make the acceptance of the project smooth.